pkg/types/chain_reader.go

```diff
@@ -45,37 +46,40 @@ type ChainReader interface {
 	// Note that implementations should ignore extra fields in params that are not expected in the call to allow easier
 	// use across chains and contract versions.
 	// Similarly, when using a struct for returnVal, fields in the return value that are not on-chain will not be set.
-	GetLatestValue(ctx context.Context, contractName, method string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error
+	GetLatestValue(ctx context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error
 
 	// BatchGetLatestValues batches get latest value calls based on request, which is grouped by contract names that each have a slice of BatchRead.
 	// BatchGetLatestValuesRequest params and returnVal follow same rules as GetLatestValue params and returnVal arguments, with difference in how response is returned.
 	// BatchGetLatestValuesResult response is grouped by contract names, which contain read results that maintain the order from the request.
 	// Contract call errors are returned in the Err field of BatchGetLatestValuesResult.
 	BatchGetLatestValues(ctx context.Context, request BatchGetLatestValuesRequest) (BatchGetLatestValuesResult, error)
 
-	// Bind will override current bindings for the same contract, if one has been set and will return an error if the
-	// contract is not known by the ChainReader, or if the Address is invalid
+	// Bind will add provided bindings and will return an error if the contract is not known by the ContractReader, or if
+	// the Address is invalid. Any provided binding that already exists should result in a noop.
 	Bind(ctx context.Context, bindings []BoundContract) error
 
+	// Unbind will remove all bindings provided.
```

Comment:
nit, does "Unbind will remove all provided bindings." sound better?

Reply:
6 of one, half dozen of another. Every english/literature teacher in my past would be appalled at my writing. lol